Update miscellaneous.md (#170)
common-workflow-language/user_guide: _extras/miscellaneous.md

    type: File
    outputSource: first/txt
```
### Optional Inputs 💯

> In CWL unlike WDL optional input settings are defined in the type section of the yaml and not on the declaration. Notice the question mark is on the type.

Reviewer: ```suggestion
To make an input parameter optional, add a question mark to the type declaration.
```

```yaml
inputs:
  InputRead1:
    type: File
    inputBinding:
      position: 100

  #Optional Inputs
  isCasava:
    type: boolean?
    inputBinding:
      position: 1
      prefix: "--casava"
```
<a name="enuminputs"></a>

> ### Enum Inputs ⚜️

Reviewer: This is defined at https://www.commonwl.org/user_guide/19-custom-types/index.html

Maybe it deserves its own mini-lesson?

Author: I would say a mini-lesson would be good, this was a boiled down version for all of us to reference quickly.

        name: genome_bam
        fields:
          genomebam:
            type: boolean
            inputBinding:
              prefix: "--genomebam"

          gtf:
            type: File
            inputBinding:
              prefix: "--gtf"

          chromosomes:
            type: File
            inputBinding:
              prefix: "--chromosomes"
```
### Setting Mutually Exclusive Parameters

In order to properly set fields in a record input type, you need to pass a dictionary to the input to properly set the parameters. This is done by using inline javascript and returning the dictionary with the key of the field you want to set. The source field is set to indicate the input from the workflow to be used as the value. 

```yaml
steps:

  build_hisat2_index:
    run: ../Tools/Hisat2-Index.cwl
    in:
      InputFiles:
        source: FastaFiles
        valueFrom : | 
          ${return {"fasta": self};}

      IndexName: IndexName
    
    out: [indexes]
```

### Setting Booleans

These can be set by using the default field
```yaml
input:
  default:true
```
### Concating Strings in Inputs

The valueFrom field must be used instead of default.

```yaml
input:

>   valueFrom: $("My String:" + input.stringvalue)

Reviewer: ```suggestion
  valueFrom: |
     My String: $(input.stringvalue)
```
